docs/index.md

```diff
@@ -167,131 +167,177 @@ Returning:
+  # name without using a string.
+  renders_many :post, PostComponent
 
-##### Example ViewComponent with Slots
+  class HeaderComponent < ViewComponent::Base
+    attr_reader :title
 
-`# box_component.rb`
+    def initialize(title:)
+    end
+  end
+end
+```
 
-```ruby
-class BoxComponent < ViewComponent::Base
-  include ViewComponent::Slotable
+`# blog_component.html.erb`
 
-  with_slot :body, :footer
-  with_slot :header, class_name: "Header"
-  with_slot :row, collection: true, class_name: "Row"
+```erb
+<div>
+  <h1><%= header %></h1> <!-- render the header component -->
 
-  class Header < ViewComponent::Slot
-    def initialize(classes: "")
-      @classes = classes
-    end
+  <% posts.each do |post| %>
+    <div class="blog-post-wrapper">
+      <%= post %> <!-- render an individual post -->
+    </div>
+  <% end %>
+</div>
+```
+
+`# index.html.erb`
+
+```erb
+<%= render BlogComponent.new do |c| %>
+  <% c.header do %>
+    <%= link_to "My Site", root_path %>
+  <% end %>
+
+  <%= c.post(title: "My blog post") do %>
+    Really interesting stuff.
+  <% end %>
+
+  <%= c.post(title: "Another post!") do %>
+    Blog every day.
+  <% end %>
+<% end %>
+```
+
+##### Labmda Slots
 
-    def classes
-      "Box-header #{@classes}"
+Lambda slots render their return value on the page. This is useful for creating
+small slots via helpers like `content_tag` or using returning components with
+parent-component specific default values.
+
+```ruby
+class Blogcomponent < ViewComponent::Base
+  # Renders the returned string
+  renders_one :header, -> (title:) do
+    content_tag :h1 do
+      link_to "My Blog", root_path
     end
   end
 
-  class Row < ViewComponent::Slot
-    def initialize(theme: :gray)
-      @theme = theme
-    end
+  # Returns a component that will be rendered in that slot with a default
+  # argument.
+  renders_many :posts, -> (title:, classes:) do
+    PostComponent.new(title: title, classes: "my-default-class " + classes)
+  end
+end
+```
 
-    def theme_class_name
-      case @theme
-      when :gray
-        "Box-row--gray"
-      when :hover_gray
-        "Box-row--hover-gray"
-      when :yellow
-        "Box-row--yellow"
-      when :blue
-        "Box-row--blue"
-      when :hover_blue
-        "Box-row--hover-blue"
-      else
-        "Box-row--gray"
-      end
+##### Pass Through Slots
+
+Pass through slots simply render the content given the block.
+
+Pass through slots can be defined by omitting the second argument to
+`renders_one` and `renders_many`.
+
+`# blog_component.rb`
+
+```ruby
+class BlogComponnt < ViewComponent::Base
+  renders_one :header
+end
+```
+
+`# index.html.erb`
+
+```ruby
+<div>
+  <%= render BlogComponent.new do |c| %>
+    <%= c.header do %>
+      <%= link_to "My blog", root_path %>
+    <% end %>
+  <% end %>
+</div>
+```
+
+##### Rendering Collections
+
+Slots defined with `renders_many` can be used to render a collection.
```

Comment:
```suggestion
Collection slots (declared with `renders_many`) can also be passed a collection.
```

Reply:
While we cover it in the section on delegate slots, I think we should explicitly document the single item setter (`c.link`) for collection slots somewhere.